zerver/lib/topic.py

```diff
@@ -321,3 +321,9 @@ def participants_for_topic(realm_id: int, recipient_id: int, topic_name: str) ->
         ).values_list("id", flat=True)
     )
     return participants
+
+
+def maybe_rename_general_chat_to_empty_topic(topic_name: str) -> str:
+    if topic_name == "general chat":
+        topic_name = ""
+    return topic_name
```

Comment:
I'm not sure we want this behavior here; it should be in the `do_send_message` code path only, only for clients that don't support the `""` topic. We don't want to impact clients that do understand the modern general chat format and want to send to "general chat" as a topic, if we can avoid it.

Reply:
Hmm, I guess what's awkward is that the "send message" API request code path does not at present accept an `empty_topic_name` client capability parameter. But probably we could make it do that?

Reply:
Having now read the branch more fully, I think you've taken the right approach; too many other code paths need to deal with this for us to avoid sacrificing "general chat" as a valid topic during the transitional period. 

We will probably want to move relatively quickly on this transition, though, in terms of adding explicit mobile app support for the client capability, though, since this does feel like a messy wart to live with.